api/src/http/auth.rs

```diff
@@ -0,0 +1,127 @@
+use crate::http::{AppState, Error, GitHubUser, Result};
+use axum::extract::{Json, State};
+
+use serde::{Deserialize, Serialize};
+use sqlx::PgPool;
+
+#[derive(Serialize, Deserialize)]
+pub struct AuthTokenPayload {
+    pub access_token: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct RegisterResponse {
+    pub message: String,
+}
+
+pub async fn register(
+    State(state): State<AppState>,
+    Json(payload): Json<AuthTokenPayload>,
+) -> Result<Json<RegisterResponse>> {
+    let AuthTokenPayload { access_token } = payload;
+    let AppState { db, client, .. } = state;
```

Comment:
Avoid destructuring, if you write like this: `&state.db`, ​​you spend much less lines.